File: Backend/Events/Converters/Parties/DigimonSlotConverter.cs
using Backend.Domain.Models.Parties;
using Backend.Events.DTO.Party;

namespace Backend.Events.Converters.Parties;

public static class DigimonSlotConverter
{
    public static DigimonSlotDTO ToDTO(DigimonSlot slot) => new()
    {
        Index = slot.Index,
        DigimonId = slot.DigimonId,
        Digimon = slot.Digimon != null ? DigimonConverter.ToDTO(slot.Digimon) : null
    };
}

File: Frontend/src/converters/DigimonSlotConverter.ts
import type { DigimonSlotDTO } from '../events/dto/Parties/DigimonSlotDTO';
import type { Digimon } from '../models/Digimon';
import { DigimonConverter } from './DigimonConverter';

export class DigimonSlotConverter {
    public static convert(slotDto: DigimonSlotDTO | null): Digimon | null {
        if (!slotDto || !slotDto.digimon) return null;
        return DigimonConverter.convert(slotDto.digimon, slotDto.index);
    }
}
